nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentJsonWriter.java

```diff
@@ -47,13 +55,22 @@
 @Setup(mode = SINGLETON, priority = REFERENCE)
 public class CommentJsonWriter extends ExtensibleEntityJsonWriter<Comment> {
 
+    public static final String FETCH_REPLIES_SUMMARY = "repliesSummary";
+
+    @Inject
+    protected CommentManager commentManager;
+
     public CommentJsonWriter() {
         super(COMMENT_ENTITY_TYPE, Comment.class);
     }
 
     @Override
     protected void writeEntityBody(Comment entity, JsonGenerator jg) throws IOException {
         writeCommentEntity(entity, jg);
+        boolean includeRepliesSummary = ctx.getFetched(COMMENT_ENTITY_TYPE).contains(FETCH_REPLIES_SUMMARY);
```

Comment:
I am not sure why the `writeCommentEntity` is `static`, it shouldn't IMO. And the `includeRepliesSummary` fetch thing would go inside.

Reply:
Other classes are relying on this method (e.g. AnnotationJsonWriter) and, for this reason, I won't change it for now, but would be interesting to have some feedback from Team Core (@funshodavid)

Reply:
Indeed, since annotation is an extension of comment it shares common code.